bin/plugin/commands/readme.js

```diff
@@ -0,0 +1,177 @@
+	},
+	{
+		argname: '-t, --token <token>',
+		description: 'GitHub token',
+	},
+];
+
+/**
+ * Command that updates the `readme.txt` file.
+ *
+ * @param {WPReadmeCommandOptions} opt
+ */
+exports.handler = async ( opt ) => {
+	await updateReadme( {
+		owner: config.githubRepositoryOwner,
+		repo: config.githubRepositoryName,
+		directory: opt.directory || 'modules',
+		milestone: opt.milestone,
+		token: opt.token,
+	} );
+};
+
+/**
+ * Returns a promise resolving to the module description list string for the `readme.txt` file.
+ *
+ * @param {WPReadmeSettings} settings Readme settings.
+ *
+ * @return {Promise<string>} Promise resolving to module description list in markdown, with trailing newline.
+ */
+async function getModuleDescriptionList( settings ) {
+	const moduleDescriptions = await getModuleDescriptions(
+		settings.directory
+	);
+
+	return moduleDescriptions
+		.map( ( moduleData ) => {
+			return `* **${ moduleData.name }:** ${ moduleData.description }`;
+		} )
+		.join( '\n' )
+		.concat( '\n' );
+}
+
+/**
+ * Updates the `readme.txt` file with the given module description list.
+ *
+ * @param {string} moduleList Module description list in markdown, with trailing newline.
+ */
+function updateReadmeModuleDescriptionList( moduleList ) {
+	const readmeFile = path.join( '.', 'readme.txt' );
+	const fileContent = fs.readFileSync( readmeFile, 'utf8' );
+	const newContent = fileContent.replace(
+		/(the following performance modules:\s+)((\*.*\n)+)/,
+		( match, prefix ) => {
+			return `${ prefix }${ moduleList }`;
+		}
+	);
+	fs.writeFileSync( readmeFile, newContent );
+}
+
+/**
+ * Updates the `readme.txt` file with the given changelog.
+ *
+ * @param {string}           changelog Changelog in markdown, with trailing newline.
+ * @param {WPReadmeSettings} settings  Readme settings.
+ */
+function updateReadmeChangelog( changelog, settings ) {
+	const regex = new RegExp( `= ${ settings.milestone } =[^=]+` );
+
+	const readmeFile = path.join( '.', 'readme.txt' );
+	const fileContent = fs.readFileSync( readmeFile, 'utf8' );
+
+	let newContent;
+	if ( fileContent.match( regex ) ) {
+		newContent = fileContent
+			.replace( regex, changelog )
+			.trim()
+			.concat( '\n' );
+	} else {
+		newContent = fileContent.replace(
+			/(== Changelog ==\n\n)/,
+			( match ) => {
+				return `${ match }${ changelog }`;
+			}
+		);
+	}
+	fs.writeFileSync( readmeFile, newContent );
+}
+
+/**
+ * Updates the `readme.txt` file with latest module description list and optionally a specific release changelog.
+ *
+ * @param {WPReadmeSettings} settings Readme settings.
+ */
+async function updateReadme( settings ) {
+	if ( ! settings.milestone ) {
+		log(
+			formats.title(
+				`\n💃Updating readme.txt for "${ settings.directory }"\n\n`
+			)
+		);
+	} else {
+		log(
+			formats.title(
+				`\n💃Updating readme.txt for "${ settings.directory }" and changelog for milestone "${ settings.milestone }"\n\n`
+			)
+		);
+	}
```

Comment:
Probably a personal, preference but inverting the logic on this conditional would remove the need to have a `!` operator.


```suggestion
	if ( settings.milestone ) {
		log(
			formats.title(
				`\n💃Updating readme.txt for "${ settings.directory }" and changelog for milestone "${ settings.milestone }"\n\n`
			)
		);
	} else {
		log(
			formats.title(
				`\n💃Updating readme.txt for "${ settings.directory }"\n\n`
			)
		);
	}
```